### PR TITLE
Add ConsensusParams updates handling logic

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
## Overview
Closes: #325

Reference to corresponding [cometBFT code](https://github.com/cometbft/cometbft/blob/6c1deb50b146bf8d27a65fbce8dbba89052dd607/internal/state/execution.go#L615).

Removes validator updates code as well that is now unnecessary due to switch to single sequencer.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Comment by Manav-Aggarwal** [comment]
Blocked on migration of main branch to `v0.50.x` so lint CI and testing CI fails currently but feel free to review.

File: types/params.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import (
+	"errors"
+	"fmt"
+
+	cmtypes "github.com/cometbft/cometbft/types"
+)
+
+// ConsensusParamsValidateBasic validates the ConsensusParams to ensure all values are within their
+// allowed limits, and returns an error if they are not.
+func ConsensusParamsValidateBasic(params cmtypes.ConsensusParams) error {
```
Comment (MSevey) [inline types/params.go:12]: this should have a test to probe each of the checks.

File: types/serialization.go [hunk 2]
```diff
@@ -335,3 +336,23 @@ func byteSlicesToSignatures(bytes [][]byte) []Signature {
 	}
 	return sigs
 }
+
+// ConsensusParamsFromProto converts protobuf consensus parameters to consensus parameters
+func ConsensusParamsFromProto(pbParams cmproto.ConsensusParams) types.ConsensusParams {
```
Comment (MSevey) [inline types/serialization.go:341]: this could also have a quick test

**Review by MSevey — CHANGES_REQUESTED** [review]
lgtm just 2 quick tests

**Review by tzdybal — APPROVED** [review]
LGTM.

**Review by MSevey — APPROVED** [review]

**Pull request merged**